src/lib/services/workflow-service.ts

```diff
@@ -363,18 +379,53 @@ export async function resetWorkflow({
 
   const body: Replace<
     ResetWorkflowRequest,
-    { workflowTaskFinishEventId: string; resetReapplyType: ResetReapplyType }
+    { workflowTaskFinishEventId: string }
   > = {
     workflowExecution: {
       workflowId,
       runId,
     },
     workflowTaskFinishEventId: eventId,
-    resetReapplyType: reapplyType,
     requestId: v4(),
     reason: formattedReason,
   };
 
+  if (get(isCloud) || minimumVersionRequired('1.24.0', get(temporalVersion))) {
+    const resetReapplyExcludeTypes: ResetWorkflowRequest['resetReapplyExcludeTypes'] =
+      [];
+
+    if (!excludeSignals && !excludeUpdates) {
+      resetReapplyExcludeTypes.push(
+        ResetReapplyExcludeType.RESET_REAPPLY_EXCLUDE_TYPE_UNSPECIFIED,
+      );
+    }
+
+    if (excludeSignals) {
+      resetReapplyExcludeTypes.push(
+        ResetReapplyExcludeType.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL,
+      );
+    }
+
+    if (excludeUpdates) {
+      resetReapplyExcludeTypes.push(
+        ResetReapplyExcludeType.RESET_REAPPLY_EXCLUDE_TYPE_UPDATE,
+      );
+    }
+
+    body.resetReapplyExcludeTypes = resetReapplyExcludeTypes;
+    body.resetReapplyType = ResetReapplyType.RESET_REAPPLY_TYPE_ALL_ELIGIBLE;
```

Comment:
It might be worth leaving a comment in the code saying that in the future we won't _need_ to explicitly send `body.resetReapplyType`, since `ResetReapplyType.RESET_REAPPLY_TYPE_ALL_ELIGIBLE` will be the server-side default. AIUI that will become true for server >= 1.25.1 (see https://github.com/temporalio/temporal/pull/5956)

But to be clear, this code is correct as-is.